team-fingerprints-web/src/components/OnboardingProfileModal/__tests__/OnboardingProfileModal.spec.tsx

```diff
@@ -0,0 +1,66 @@
+import OnboardingProfileModal from "../OnboardingProfileModal";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { AppSetup } from "testing/helpers/AppSetup";
+import userEvent from "@testing-library/user-event";
+import { createMemoryHistory, MemoryHistory } from "history";
+
+const setVisibleProfileModalMock = jest.fn();
+
+interface RenderComponentProps {
+  history?: MemoryHistory;
+  visibleProfileModal?: boolean;
+}
+
+const renderComponent = ({
+  history,
+  visibleProfileModal = true,
+}: RenderComponentProps) => {
+  render(
+    <AppSetup history={history}>
+      <OnboardingProfileModal
+        visibleProfileModal={visibleProfileModal}
+        setVisibleProfileModal={setVisibleProfileModalMock}
+        surveyId="100"
+      />
+    </AppSetup>
+  );
+};
+
+describe("OnboardingProfileModal", () => {
+  it("does show modal if visibleProfileModal is true", () => {
+    renderComponent({});
+
+    expect(screen.getByRole("dialog")).toBeInTheDocument();
+  });
+
+  it("does not show modal if visibleProfileModal is false", () => {
+    renderComponent({ visibleProfileModal: false });
+
+    expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
+  });
+
+  it("does navigate with correct props and trigger setVisibleProfileModal when click Go To Profile button", async () => {
+    const history = createMemoryHistory({ initialEntries: ["/"] });
+
+    renderComponent({ history });
+
+    const goToProfileButton = screen.getByText("Go to profile");
+    await userEvent.click(goToProfileButton);
+
+    expect(setVisibleProfileModalMock).toHaveBeenCalledWith(false);
+
+    expect(history.location.pathname).toEqual("/profile");
+    expect(history.location.state).toEqual({ surveyId: "100" });
+  });
+
+  it("does not close modal if click outside or press escape", async () => {
+    renderComponent({});
+
+    await userEvent.click(document.body);
+    fireEvent.keyPress(global.document, {
+      key: "Escape",
+    });

```

Comment:
You can use `userEvent` for that too
```suggestion
    await userEvent.keyboard("{esc}");
```

Reply:
Right, done